AsyncDisplayKit/Details/ASChangeSetDataController.mm

```diff
@@ -205,6 +209,8 @@ - (void)moveRowAtIndexPath:(NSIndexPath *)indexPath toIndexPath:(NSIndexPath *)n
   ASDisplayNodeAssertMainThread();
   [self beginUpdates];
   [_changeSet moveItemFromIndexPath:indexPath toIndexPath:newIndexPath animationOptions:animationOptions];
+  [_changeSet deleteItems:@[indexPath] animationOptions:animationOptions];
+  [_changeSet insertItems:@[newIndexPath] animationOptions:animationOptions];
   [self endUpdates];
```

Comment:
I made use of the original idea of split here, reusing the existing `delete` + `insert` logic.